etl/tests/pipeline_with_partitioned_table.rs

```diff
@@ -0,0 +1,1400 @@
+    let parent_inserts = grouped
+        .get(&(EventType::Insert, parent_table_id))
+        .cloned()
+        .unwrap_or_default();
+    assert_eq!(parent_inserts.len(), 1);
+}
+
+/// Tests that detaching and dropping a partition does not emit DELETE or TRUNCATE events.
+/// Partition management is a DDL operation, not DML, so no data events should be generated.
+#[tokio::test(flavor = "multi_thread")]
+async fn partition_drop_does_not_emit_delete_or_truncate() {
+    init_test_tracing();
+    let database = spawn_source_database().await;
+
+    let table_name = test_table_name("partitioned_events_drop");
+    let partition_specs = [("p1", "from (1) to (100)"), ("p2", "from (100) to (200)")];
+
+    let (parent_table_id, _partition_table_ids) =
+        create_partitioned_table(&database, table_name.clone(), &partition_specs)
+            .await
+            .unwrap();
+
+    database
+        .run_sql(&format!(
+            "insert into {} (data, partition_key) values ('event1', 50), ('event2', 150)",
+            table_name.as_quoted_identifier()
+        ))
+        .await
+        .unwrap();
+
+    let publication_name = "test_partitioned_pub_drop".to_string();
+    database
+        .create_publication(&publication_name, std::slice::from_ref(&table_name))
+        .await
+        .unwrap();
+
+    let state_store = NotifyingStore::new();
+    let destination = TestDestinationWrapper::wrap(MemoryDestination::new());
+
+    let parent_sync_done = state_store
+        .notify_on_table_state_type(parent_table_id, TableReplicationPhaseType::SyncDone)
+        .await;
+
+    let pipeline_id: PipelineId = random();
+    let mut pipeline = create_pipeline(
+        &database.config,
+        pipeline_id,
+        publication_name,
+        state_store.clone(),
+        destination.clone(),
+    );
+
+    pipeline.start().await.unwrap();
+    parent_sync_done.notified().await;
+
+    let events_before = destination.get_events().await;
+    let grouped_before = group_events_by_type_and_table_id(&events_before);
+    let delete_count_before = grouped_before
+        .get(&(EventType::Delete, parent_table_id))
+        .map(|v| v.len())
+        .unwrap_or(0);
+    let truncate_count_before = grouped_before
+        .get(&(EventType::Truncate, parent_table_id))
+        .map(|v| v.len())
+        .unwrap_or(0);
+
+    // Detach and drop one child partition (DDL should not generate DML events).
+    let partition_p1_name = format!("{}_{}", table_name.name, "p1");
+    let partition_p1_qualified = format!("{}.{}", table_name.schema, partition_p1_name);
+    database
+        .run_sql(&format!(
+            "alter table {} detach partition {}",
+            table_name.as_quoted_identifier(),
+            partition_p1_qualified
+        ))
+        .await
+        .unwrap();
+    database
+        .run_sql(&format!("drop table {partition_p1_qualified}"))
+        .await
+        .unwrap();
+
+    let _ = pipeline.shutdown_and_wait().await;
```

Comment:
To test for an absence of events, we should insert one row into an existing partition after detaching & dropping the partition table and wait for that event to arrive. In the current form, the test could be passing because we shutdown the pipeline too quickly.